src/chrome/internal/sources/sourceTextRetriever.ts

```diff
@@ -19,28 +19,29 @@ import { printArray } from '../../collections/printing';
  */
 @injectable()
 export class SourceTextRetriever {
-    private _sourceToText = new ValidatedMap<ILoadedSource, string>();
+    private _sourceToText = new ValidatedMap<ILoadedSource, Promise<string>>();
 
     constructor(@inject(TYPES.IScriptSources) private readonly _scriptSources: IScriptSourcesRetriever) { }
 
-    public async text(loadedSource: ILoadedSource): Promise<string> {
+    // We want this method to be sync, so if we get 2 simultaneous calls, the second call will return the promise of the first call
```

Comment:
This method still returns a promise, I don't think there is a behavior change here?

Reply:
Oh, now when there are two calls you will set the promise in the map instead of calling getScriptSource twice, got it. Don't call it sync though

Reply:
Changed to:
    // We want this method to add an entry to the map this._sourceToText atomically, so if we get 2 simultaneous calls,
    // the second call will return the promise/result of the first call